File: 2015/js/day1/main.js
"use strict";
let directions = require("fs").readFileSync("./directions.txt", { encoding: "utf-8" }).trim();

let floor = 0, step = 1, wentDown = false;
for(let char of directions) {
	floor += (char === "(" ? 1 : -1);
	if(floor < 0 && !wentDown) {
		wentDown = true;
		console.log(`Went below ground at step ${step}`);
	}
	step++;
}

console.log(`Final floor is ${floor}`);

File: 2015/python/day1/main.py
directions = open("directions.txt", "r");
floor, step, hasCrossed = 0, 1, False

direction = directions.read(1)
while direction != "":
	if direction == "(":
		floor = floor + 1
	elif direction == ")":
		floor = floor - 1
	if floor < 0 and hasCrossed == False:
		print("Part 2: Crossed over on step {0}".format(step))
		hasCrossed = True
	step = step + 1
	direction = directions.read(1)

print("Part 1: Ended up on floor {0}".format(floor))
